### PR TITLE
check for access conditions and link to UV

### PR DESCRIPTION
**Pull request by gestchild**
We aren't yet set up to display works that have access conditions on them, so we pass them to the universal viewer on the library site for now.

### CODE REVIEW

File: common/utils/works.js [hunk 1]
```diff
@@ -177,6 +178,20 @@ export function getDigitalLocationOfType(
   return item;
 }
 
+export function getAccessConditionForDigialLocation(
```
Comment (jamesgorrie) [inline common/utils/works.js:181]: Should we be writing tests for this?

File: catalogue/webapp/components/WorkDetails/WorkDetails.js [hunk 2]
```diff
@@ -256,89 +296,78 @@ const WorkDetails = ({
...
+                    Contains:{' '}
+                    {childManifestsCount > 0
+                      ? `${childManifestsCount} volumes`
+                      : imageCount > 0
+                      ? `${imageCount} ${imageCount === 1 ? 'image' : 'images'}`
+                      : ''}
+                  </p>
+                </Space>
+              )}
+            </>
```
Comment (jamesgorrie) [inline catalogue/webapp/components/WorkDetails/WorkDetails.js:370]: With all these pieces of logic, could we write a method that spits out the different know states and use those to decide what to render?

e.g.

```ts
type ItemLinkState = 'showViewer' | `showLibraryLink` | 'blah';
function getItemLinkState(stuff): ItemLinkState {
  //...
}
```

File: common/utils/works.js [hunk 3]
```diff
@@ -177,6 +178,20 @@ export function getDigitalLocationOfType(
...
+  digitalLocation: ?DigitalLocation
+): ?string {
+  if (digitalLocation) {
+    const accessConditions = digitalLocation?.accessConditions || [];
+    const accessCondition = accessConditions.find(
+      condition => condition.status
+    );
+    return accessCondition ? accessCondition.status?.id : null;
```
Comment (davidpmccormick) [inline common/utils/works.js:189]: would this be the same as `return accessCondition?.status?.id;`

File: common/utils/works.js [hunk 4]
```diff
@@ -177,6 +178,20 @@ export function getDigitalLocationOfType(
...
+  } else {
```
Comment (ishouty) [inline common/utils/works.js:190]: Could remove else and just put return null;

File: common/utils/works.js (hunk 1 above)
Comment (ishouty) [inline common/utils/works.js:181]: Digial assume you mean digital?

File: common/test/utils/works.test.js [hunk 5]
```diff
@@ -100,3 +102,48 @@ describe('getTabbableIds', () => {
...
+          type: 'AccessCondition',
+        },
+      ],
+      type: 'DigitalLocation',
+    });
+  });
+});
+
+describe('getAccessConditionForDigitalLocation', () => {
+  it('returns the status.id of the first object in an accessConditions array with a status property ', () => {
```
Comment (davidpmccormick) [inline common/test/utils/works.test.js:141]: nit: extra space at end of sentence!

**Review by davidpmccormick — APPROVED** [review]
🤘

**Pull request merged**